zhaquirks/sonoff/swv.py

```diff
@@ -0,0 +1,112 @@
+"""Sonoff SWV - Zigbee smart water valve."""
+
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+import zigpy.types as t
+from zigpy.zcl.clusters.general import (
+    Basic,
+    Identify,
+    OnOff,
+    Ota,
+    PollControl,
+    PowerConfiguration,
+    Time,
+)
+from zigpy.zcl.clusters.homeautomation import Diagnostic
+from zigpy.zcl.clusters.measurement import FlowMeasurement
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+
+SONOFF_CLUSTER_FC11_ID = 0xFC11
+SONOFF_CLUSTER_FC57_ID = 0xFC57
+ATTR_SONOFF_VALVE_STATUS = 0x500C
+
+
+class ValveStatusBitmap(t.bitmap8):
+    """Valve status value enum."""
+
+    Water_Shortage = 0x1
+    Water_Leakage = 0x2
+
+
+class SonoffFC11Cluster(CustomCluster):
+    """Sonoff manufacture specific cluster that provides valve status."""
+
+    @property
+    def _is_manuf_specific(self) -> bool:
+        """Override manufacturer specific property.
+
+        valve_status return UNSUPPORTED_ATTRIBUTE if manufacturer_specific is set on report command.
+        We therefore need to treat cluster as if it doesn't have ID within manufacturer specific range
+        and explicitly set if each attribute is manufacturer_specific or not instead in attribute list.
+        """
+        return False
+
+    cluster_id = SONOFF_CLUSTER_FC11_ID
+    ep_attribute = "sonoff_manufacturer"
+    attributes = {ATTR_SONOFF_VALVE_STATUS: ("valve_status", ValveStatusBitmap, False)}
```

Comment:
I believe attributes are deprecated. Might be better to use `ZCLAttributeDef` to define this.

Reply:
Thanks - I based this off the other Sonoff quirk, with no prior experience writing quirks. Let me look into `ZCLAttributeDef` and see how easy it is to convert over.